fuse_models/include/fuse_models/parameters/odometry_2d_params.h

```diff
@@ -89,11 +89,12 @@ struct Odometry2DParams : public ParameterBase
         fuse_core::getParamRequired(nh, "twist_target_frame", twist_target_frame);
       }
 
-      if (!differential && (!position_indices.empty() || !orientation_indices.empty()))
+      if (!position_indices.empty() || !orientation_indices.empty())
```

Comment:
This change makes the `pose_target_frame` required for the `differential` mode too.

Another option would be to skip the transformation is `pose_target_frame.empty()`, i.e. if not set (it defaults to empty). That could be a change applied to all instances of any type of target frame parameter. Thoughts?

Reply:
I like making it required.

Reply:
That's fine, but I wonder why you prefer it required as opposed to optional with the default falling back to the input message `header.frame_id`.

In a configuration with multiple odometry sources (e.g. wheel, laser, visual) we'd have to specify the `pose_target_frame` for each different odometry frame (e.g. `wheel_odom`, `laser_odom`, `visual_odom`), even for the `differential` mode with this PR. This means that any existing configuration should be updated, so the node doesn't terminate during the parameter loading step. If this is well understood and acceptable, then I'm happy to update my configurations adding the `pose_target_frame` where needed.

Reply:
I suppose I just like being explicit. We could change the parameter loading so that it does a `find`, too. Then you could have a common `pose_target_frame` that is used by all pose sources, and if you want to override it for any of them, you can.

To be clear, I don't feel strongly about this _at all_, so if you like using the "empty means skip transformation paradigm," then that's fine with me. :+1: